src/components/app/data/hooks/useEnterpriseCustomer.test.jsx

```diff
@@ -0,0 +1,95 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { QueryClientProvider } from '@tanstack/react-query';
+import { AppContext } from '@edx/frontend-platform/react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { authenticatedUserFactory } from '../services/data/__factories__';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+import { queryClient } from '../../../../utils/tests';
+import { fetchEnterpriseLearnerData } from '../services';
+import { useEnterpriseLearner } from './index';
+
+jest.mock('../services', () => ({
+  ...jest.requireActual('../services'),
+  fetchEnterpriseLearnerData: jest.fn().mockResolvedValue(null),
+}));
+const mockAuthenticatedUser = authenticatedUserFactory();
+const mockEnterpriseLearnerData = {
+  enterpriseCustomer: {
+    uuid: 'test-enterprise-customer',
+    slug: 'test-enterprise-slug',
+  },
+  enterpriseCustomerUserRoleAssignments: [],
+  activeEnterpriseCustomer: null,
+  activeEnterpriseCustomerUserRoleAssignments: [],
+  allLinkedEnterpriseCustomerUsers: [],
+  enterpriseFeatures: {},
+  staffEnterpriseCustomer: null,
+};
+
+describe('useEnterpriseCustomer', () => {
+  const Wrapper = ({ children }) => (
+    <QueryClientProvider client={queryClient()}>
+      <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
+        {children}
+      </AppContext.Provider>
+    </QueryClientProvider>
+  );
+  beforeEach(() => {
+    jest.clearAllMocks();
+    fetchEnterpriseLearnerData.mockResolvedValue(mockEnterpriseLearnerData);
+  });
+  it('should return nested hook value correctly', async () => {
+    const {
+      result,
+      waitForNextUpdate,
+    } = renderHook(() => useEnterpriseLearner(), { wrapper: Wrapper });
+    await waitForNextUpdate();
+    expect(result.current).toEqual(
+      expect.objectContaining({
+        data: mockEnterpriseLearnerData,
+      }),
+    );
+  });
+  it('should handle parent hook return value correctly', async () => {
```

Comment:
In this particular hook, there isn't really any nested hooks. It's not entirely clear why these tests couldn't also follow the pattern you started in your previous coverage part 1 PR?

```jsx
it('should return enterprise customer metadata correctly', async () => {
    const { result, waitForNextUpdate } = renderHook(() => useEnterpriseCustomer(), { wrapper: Wrapper });
    await waitForNextUpdate();
    const actualEnterpriseCustomer = result.current.data;
    expect(actualEnterpriseCustomer.uuid).toEqual(mockEnterpriseCustomer.uuid);
    expect(actualEnterpriseCustomer.slug).toEqual(mockEnterpriseCustomer.slug);
  });
  it('should return enterprise customer metadata correctly with select', async () => {
    const { result, waitForNextUpdate } = renderHook(() => useEnterpriseCustomer({
      select: (data) => data,
    }), { wrapper: Wrapper });
    await waitForNextUpdate();
    const actualEnterpriseCustomerSelectArgs = result.current.data;
    expect(actualEnterpriseCustomerSelectArgs.original).toEqual(mockEnterpriseLearnerData);
    expect(actualEnterpriseCustomerSelectArgs.transformed).toEqual(mockEnterpriseCustomer);
  });
```